src/classes/AffiliationMappingsHealthCheckVMapper.cls

```diff
@@ -0,0 +1,304 @@
+    * @description Empty constructor is not public for Singleton Pattern
+    **********************************************************************************/
+    protected AffiliationMappingsHealthCheckVMapper() {}
+
+    /**********************************************************************************
+    * @description Static method to get the current instance for Singleton pattern
+    * @return The instance of AffiliationMappingsHealthCheckVMapper.
+    **********************************************************************************/
+    public static AffiliationMappingsHealthCheckVMapper getInstance() {
+        if (instance == null) {
+            instance = new AffiliationMappingsHealthCheckVMapper();
+        }
+
+        return instance;
+    }
+
+    /**********************************************************************************
+    * @description Retrieve the results for a for an AffiliationMappings health check.
+    * @return A HealthCheckGroupVModel representing a HealtCheckDescription for Account.
+    **********************************************************************************/ 
+    public virtual HealthCheckGroupVModel getHealthCheckGroup() {
+        HealthCheckGroupVModel parentHealthCheckGroupVModel = new HealthCheckGroupVModel(
+            Label.stgHealthCheckAffiliationMappings,
+            Label.stgHealthCheckStatusPassed,
+            0,
+            0,
+            new List<String>(),
+            new List<HealthCheckItemVModel>()
+        );
+
+        Map<Id, RecordTypeModel> accRTModelsById = this.locateRecordTypeService().getModelByIdMapForSObjectType(Account.SObjectType);
+        List<AffiliationMappingsModel> affiliationMappingsModelsList = this.locateAffiliationMappingsService().getAffiliationMappingsModels();
+
+        for (AffiliationMappingsModel affiliationMappingsModel : affiliationMappingsModelsList) {
+            HealthCheckGroupVModel childAffiliationMappingsHealthCheckResults = this.getAffiliationMappingsHealthCheckResults(
+                affiliationMappingsModel,
+                accRTModelsById
+            );
+
+            // Add results from child health check group results to parent
+            parentHealthCheckGroupVModel.passedChecks += childAffiliationMappingsHealthCheckResults.passedChecks;
+            parentHealthCheckGroupVModel.totalChecks += childAffiliationMappingsHealthCheckResults.totalChecks;
+            parentHealthCheckGroupVModel.expandedRowsList.addAll(childAffiliationMappingsHealthCheckResults.expandedRowsList);
+            parentHealthCheckGroupVModel.healthCheckItemList.addAll(childAffiliationMappingsHealthCheckResults.healthCheckItemList);
+        }
+
+        parentHealthCheckGroupVModel.updateHealthCheckStatus();
+        parentHealthCheckGroupVModel.healthCheckItemList.sort();
+
+        return parentHealthCheckGroupVModel;
+    }
+
+    /**********************************************************************************
```

Comment:
Love this list! Although we will have to maintain it if we are adding more health checks.

Reply:
@xinnanlisf  I agree, I'm not sure if it's worth the maintenance, but I found it helpful for now.  I am happy to go with whatever the team decides is the best approach.